### PR TITLE
Added functionality for validation split

### LINKED ISSUE #654: Inria dataset validation set
https://project.inria.fr/aerialimagelabeling/contest/ says, "For validation, we suggest to remove the first five images of every location (e.g., austin{1-5}.tif, chicago{1-5}.tif) from the training set."

It would be nice to implement this in our Dataset, then have the option to ignore it or not in the DataModule (i.e. when you want to train with as many labels as possible).

### PR DESCRIPTION
**Pull request by Tarandeep97**
**Closes #654: Implement Inria Dataset Validation Set Split**

**Summary:**
This pull request addresses issue #654, which involves implementing the validation set split for the Inria dataset as specified in the contest guidelines (https://project.inria.fr/aerialimagelabeling/contest/).

**Changes Made:**
Added functionality to the `inria.py` file.

**Details:**
Modified the `inria.py` file to implement the validation set split. The first five images of each location can now be removed by  setting split argument to valid.

### CODE REVIEW

**Comment by Tarandeep97** [comment]
@microsoft-github-policy-service agree

**Comment by adamjstewart** [comment]
Can we call it "val" instead of "valid"? That's what most other datasets call it.

**Comment by Tarandeep97** [comment]
changed "valid" to "val", as per convention.

**Comment by adamjstewart** [comment]
Still a lot of failing tests, see https://torchgeo.readthedocs.io/en/stable/user/contributing.html for tips to fix these.

File: torchgeo/datasets/inria.py [hunk 1]
```diff
@@ -89,21 +89,28 @@ def _load_files(self, root: str) -> list[dict[str, str]]:
         Returns:
             list of dicts containing paths for each pair of image and label
         """
-        files = []
-        root_dir = os.path.join(root, self.directory, self.split)
+        train_files = []
+        val_files = []
+        test_files = []
```
Comment (adamjstewart) [inline torchgeo/datasets/inria.py:94]: What's the benefit of having 3 lists if you only ever return a single list?

File: torchgeo/datasets/inria.py [hunk 2]
```diff
@@ -72,7 +72,7 @@ def __init__(
             RuntimeError: if dataset is missing
```
Comment (adamjstewart) [inline torchgeo/datasets/inria.py:72]: Can you add this to the docstring:
```rst
.. versionchanged:: 0.5                                                          
   Added support for a *val* split.   
```

**Comment by Tarandeep97** [comment]
Hi @adamjstewart, sorry for late reply. I was occupied with my thesis project. I am working on the feedback you have provided.

**Comment by adamjstewart** [comment]
@Tarandeep97 no worries, I also just finished my thesis. Hope everything went okay!

File: torchgeo/datasets/inria.py (hunk 2 above)
Comment (Tarandeep97) [reply torchgeo/datasets/inria.py:76]: Docstring added.

**Comment by adamjstewart** [comment]
I made the regex a bit simpler and got the tests working, let me know what you think!

**Comment by Tarandeep97** [comment]
This looks perfect to me @adamjstewart . Thank you for helping!

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**